server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -457,13 +458,15 @@ onMessage authMode serverEnv wsConn msgRaw =
                            wsConn authMode params
       CMStart startMsg  -> liftIO $ onStart serverEnv wsConn startMsg
       CMStop stopMsg    -> liftIO $ onStop serverEnv wsConn stopMsg
+      -- TODO do we need onStop/removeLiveQuery cleanup here? I guess this is handled by 'onClose'?
       CMConnTerm        -> liftIO $ WS.closeConn wsConn "GQL_CONNECTION_TERMINATE received"
   where
     logger = _wseLogger serverEnv
 
 onStop :: WSServerEnv -> WSConn -> StopMsg -> IO ()
 onStop serverEnv wsConn (StopMsg opId) = do
-  -- probably wrap the whole thing in a single tx?
+  -- TODO probably wrap the whole thing in a single tx? IO in removeLiveQuery can be returned and
```

Comment:
A transaction is not really needed because there cannot be any concurrent invocations of `onMessage` on each websocket connection. In addition, `onClose` is called after the thread which invokes `onMessage` is cancelled (in `Transport.WebSocket.Server` module).